File: db-4.8.30/java/src/com/sleepycat/db/HashStats.java
/*-
 * Automatically built by dist/s_java_stat.
 * Only the javadoc comments can be edited.
 *
 * See the file LICENSE for redistribution information.
 *
 * Copyright (c) 2002-2009 Oracle.  All rights reserved.
 */

package com.sleepycat.db;

/**
The HashStats object is used to return Hash database statistics.
*/
public class HashStats extends DatabaseStats {
    // no public constructor
    /* package */ HashStats() {}

    private int hash_magic;
    /** TODO */
    public int getMagic() {
        return hash_magic;
    }

    private int hash_version;
    /** TODO */
    public int getVersion() {
        return hash_version;
    }

    private int hash_metaflags;
    /** TODO */
    public int getMetaFlags() {
        return hash_metaflags;
    }

    private int hash_nkeys;
    /** TODO */
    public int getNumKeys() {
        return hash_nkeys;
    }

    private int hash_ndata;
    /** TODO */
    public int getNumData() {
        return hash_ndata;
    }

    private int hash_pagecnt;
    /** TODO */
    public int getPageCount() {
        return hash_pagecnt;
    }

    private int hash_pagesize;
    /** TODO */
    public int getPageSize() {
        return hash_pagesize;
    }

    private int hash_ffactor;
    /** TODO */
    public int getFfactor() {
        return hash_ffactor;
    }

    private int hash_buckets;
    /** TODO */
    public int getBuckets() {
        return hash_buckets;
    }

    private int hash_free;
    /** TODO */
    public int getFree() {
        return hash_free;
    }

    private long hash_bfree;
    /** TODO */
    public long getBFree() {
        return hash_bfree;
    }

    private int hash_bigpages;
    /** TODO */
    public int getBigPages() {
        return hash_bigpages;
    }

    private long hash_big_bfree;
    /** TODO */
    public long getBigBFree() {
        return hash_big_bfree;
    }

    private int hash_overflows;
    /** TODO */
    public int getOverflows() {
        return hash_overflows;
    }

    private long hash_ovfl_free;
    /** TODO */
    public long getOvflFree() {
        return hash_ovfl_free;
    }

    private int hash_dup;
    /** TODO */
    public int getDup() {
        return hash_dup;
    }

    private long hash_dup_free;
    /** TODO */
    public long getDupFree() {
        return hash_dup_free;
    }

    /**
    For convenience, the HashStats class has a toString method
    that lists all the data fields.
    */
    public String toString() {
        return "HashStats:"
            + "\n  hash_magic=" + hash_magic
            + "\n  hash_version=" + hash_version
            + "\n  hash_metaflags=" + hash_metaflags
            + "\n  hash_nkeys=" + hash_nkeys
            + "\n  hash_ndata=" + hash_ndata
            + "\n  hash_pagecnt=" + hash_pagecnt
            + "\n  hash_pagesize=" + hash_pagesize
            + "\n  hash_ffactor=" + hash_ffactor
            + "\n  hash_buckets=" + hash_buckets
            + "\n  hash_free=" + hash_free
            + "\n  hash_bfree=" + hash_bfree
            + "\n  hash_bigpages=" + hash_bigpages
            + "\n  hash_big_bfree=" + hash_big_bfree
            + "\n  hash_overflows=" + hash_overflows
            + "\n  hash_ovfl_free=" + hash_ovfl_free
            + "\n  hash_dup=" + hash_dup
            + "\n  hash_dup_free=" + hash_dup_free
            ;
    }
}

File: db-4.8.30/csharp/HashStats.cs
/*-
 * See the file LICENSE for redistribution information.
 *
 * Copyright (c) 2009 Oracle.  All rights reserved.
 *
 */
using System;
using System.Collections.Generic;
using System.Text;

namespace BerkeleyDB {
    /// <summary>
    /// Statistical information about a HashDatabase
    /// </summary>
    public class HashStats {
        private Internal.HashStatStruct st;
        internal HashStats(Internal.HashStatStruct stats) {
            st = stats;
        }

        /// <summary>
        /// Number of big key/data pages. 
        /// </summary>
        public uint BigPages { get { return st.hash_bigpages; } }
        /// <summary>
        /// Bytes free on big item pages. 
        /// </summary>
        public ulong BigPagesFreeBytes { get { return st.hash_big_bfree; } }
        /// <summary>
        /// Bytes free on bucket pages. 
        /// </summary>
        public ulong BucketPagesFreeBytes { get { return st.hash_bfree; } }
        /// <summary>
        /// Number of dup pages. 
        /// </summary>
        public uint DuplicatePages { get { return st.hash_dup; } }
        /// <summary>
        /// Bytes free on duplicate pages. 
        /// </summary>
        public ulong DuplicatePagesFreeBytes { get { return st.hash_dup_free; } }
        /// <summary>
        /// Fill factor specified at create. 
        /// </summary>
        public uint FillFactor { get { return st.hash_ffactor; } }
        /// <summary>
        /// Pages on the free list. 
        /// </summary>
        public uint FreePages { get { return st.hash_free; } }
        /// <summary>
        /// Metadata flags. 
        /// </summary>
        public uint MetadataFlags { get { return st.hash_metaflags; } }
        /// <summary>
        /// Magic number. 
        /// </summary>
        public uint MagicNumber { get { return st.hash_magic; } }
        /// <summary>
        /// Number of data items. 
        /// </summary>
        public uint nData { get { return st.hash_ndata; } }
        /// <summary>
        /// Number of hash buckets. 
        /// </summary>
        public uint nHashBuckets { get { return st.hash_buckets; } }
        /// <summary>
        /// Number of unique keys. 
        /// </summary>
        public uint nKeys { get { return st.hash_nkeys; } }
        /// <summary>
        /// Number of overflow pages. 
        /// </summary>
        public uint OverflowPages { get { return st.hash_overflows; } }
        /// <summary>
        /// Bytes free on ovfl pages. 
        /// </summary>
        public ulong OverflowPagesFreeBytes { get { return st.hash_ovfl_free; } }
        /// <summary>
        /// Page count. 
        /// </summary>
        public uint nPages { get { return st.hash_pagecnt; } }
        /// <summary>
        /// Page size. 
        /// </summary>
        public uint PageSize { get { return st.hash_pagesize; } }
        /// <summary>
        /// Version number. 
        /// </summary>
        public uint Version { get { return st.hash_version; } }

    }
}